gapis/replay/wait_for_perfetto.go

```diff
@@ -0,0 +1,41 @@
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapir"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/trace"
+)
+
+func WaitForPerfetto(traceOptions *service.TraceOptions, h *SignalHandler) *WaitForFence {
+	t := WaitForFence{}
+	tcb := func(ctx context.Context, p *gapir.FenceReadyRequest) {
+		go func() {
+			trace.Trace(ctx, traceOptions.Device, h.StartSignal, h.StopSignal, h.ReadyFunc, traceOptions, &h.Written)
+			if !h.DoneSignal.Fired() {
+				h.DoneFunc(ctx)
+			}
+		}()
+		h.ReadySignal.Wait(ctx)
+	}
+	t.transformCallback = tcb
```

Comment:
instead of assigning to the fields, just create the `WaitForFence` instance at the end.